### PR TITLE
[TASK] Remove unprocessable tags from the DOM, not from the raw HTML

### PR DESCRIPTION
**Pull request by oliverklee**
Required for #554.

### CODE REVIEW

**Comment by oliverklee** [comment]
This PR will conflict with #626. After any one of those two has been merged, I'll take care of rebasing the other one.

**Comment by oliverklee** [comment]
Rebased and repushed.

File: src/Emogrifier.php [hunk 1]
```diff
@@ -412,8 +412,7 @@ private function createRawDomDocument($html)
      */
```
Comment (zoliszabo) [inline src/Emogrifier.php]: The description of this method is not accurate anymore. It should be something like:
```
     * Returns the HTML with added document type and Content-Type meta tag if needed.
```
Given this, I am not sure that `unifyHtml` is the best name anymore for this method. What do you think?

Comment (oliverklee) [reply src/Emogrifier.php]: Good catch! I've updated the descriptions and will repush after we've come to a conclusion about the name.

Comment (oliverklee) [reply src/Emogrifier.php]: The name should communicate that the method prepares the HTML in a way that it is safe for creating a DOM document from.

What do you think about the name `prepareHtmlForDomConversion`?

Comment (zoliszabo) [reply src/Emogrifier.php]: I like it!

Comment (oliverklee) [reply src/Emogrifier.php]: Great! Renamed and repushed.

**Pull request merged**